lib/core/scheduler.rb

```diff
@@ -90,6 +97,7 @@ def sync_triggered?(connector_settings)
         Utility::Cron.quartz_to_crontab(current_schedule)
       rescue StandardError => e
         Utility::ExceptionTracking.log_exception(e, "Unable to convert quartz (#{current_schedule}) to crontab.")
+        return false
```

Comment:
I think this is a bug. If the quartz can't be parsed correctly, we should not sync.

Reply:
But we do not sync as far as I remember, is it not the case?

Reply:
We do not sync, because in the next step, `cron_parser` is `nil`. But I think we shouldn't go to the next step at all.

https://github.com/elastic/connectors-ruby/blob/93403cc4c8a568e4b1c65df7a8e357cad5b05f8a/lib/core/scheduler.rb#L102-L108

Reply:
I don't understand though, next step after what?

Reply:
Here is the code before this PR:
https://github.com/elastic/connectors-ruby/blob/18082d7e652d89f69f692891f5ff00c5861ea8e4/lib/core/scheduler.rb#L89-L100

Because there's no `return false` after line 92, it goes to line 94 with `current_schedule = nil`, which results `cron_parser = nil`. In the end, this method `returns false` at line 99.

But we should not reach line 94 as we know `current_schedule` is `nil`.